vars/withGoEnv.groovy

```diff
@@ -44,3 +45,9 @@ def call(Map args = [:], Closure body) {
     }
   }
 }
+
+def checkGoPath(){
+  if(fileExists(file: "${env.GOPATH}/go.mod")){
```

Comment:
is `GOPATH` defined earlier? If so, where?

Reply:
it is defined by `withGoEnv`

Reply:
right, but it does not exist yet when the checkGoPath is called

Reply:
I've moved the call inside the `withGoEnvUnix` and `withGoEnvWindows`  closures